lib/puppet/transaction/report.rb

```diff
@@ -386,6 +394,89 @@ def self.default_format
 
   private
 
+  # Using the last applied values from our persistence store, and the current
+  # report data, iterate across all events finding any corrective_change that
+  # might exist and marking the event, resource and report appropriately.
+  def calculate_corrective_change
+    persistence = Puppet::Transaction::Persistence.new
```

Comment:
This function handles loading persisted data, calculating corrective change, and then saving the persisted data again - I would rather we have the actual data processing logic in a separate function and another function that handles the persistence loading/saving and invoking the data processing logic.